src/Marten/Services/SprocCall.cs

```diff
@@ -90,6 +88,9 @@ public SprocCall JsonBodies(string argName, ArraySegment<char>[] bodies)
 
         public SprocCall Param(string argName, object value, NpgsqlDbType dbType)
         {
+            if ((value is Enum || value is Guid) && dbType == NpgsqlDbType.Varchar)
```

Comment:
Same thing here about the enum configuration on the serializer

Reply:
Here it's the same case as I described with Array of Guids. It seems that there is no more automatic conversion here. They provided some changes to automatically selecting the handlers, but I don't know precisely, because it's a lack of documentation.